### PR TITLE
GH-543 rdfxml rdf parse type literal

### PR DESCRIPTION
**Pull request by deepali17043**
GitHub issue resolved: #543  

Briefly describe the changes proposed in this PR:

According to the suggestions on our PR #2280 we have cherry-picked the commits related to this issue and made the required changes.
---- 
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/master/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change
 - [x] every commit has been [signed off](https://stackoverflow.com/questions/1962094/what-is-the-sign-off-feature-in-git-for)

Note: we merge all feature pull requests using [squash and merge](https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits). See [RDF4J git merge strategy](https://rdf4j.org/documentation/developer/merge-strategy/) for more details.

### CODE REVIEW

**Comment by deepali17043** [comment]
Including the RDFXML Parser Test requires the private class `CanonXMLValueFactory` and we are unable to access it in `RDFXMLWriterTest`
Please suggest what can be done.

**Comment by jeenbroekstra** [comment]
> Including the RDFXML Parser Test requires the private class `CanonXMLValueFactory` and we are unable to access it in `RDFXMLWriterTest`
> Please suggest what can be done.

Apologies, I may have confused things by suggesting you should add a test to RDFXMLWriterTest - that should of course be RDFXMLParserTest! After all, the fix you did is a fix in the parser, not in the writer. 

So, to make it clear:

1. add one or two new test methods to `org.eclipse.rdf4j.rio.rdfxml.RDFXMLParserTest`
2. in these methods, read a file that contains a value with parse type literal - there are some existing tests there that read a file, the actual files can be found in `src/test/resources`.
3. parse this file with the RDFXMLParser, and then check  (using `assertTrue` or `assertEquals`) that the actual literal value that the parser produces is as expected. You can easily do this by just having a separate String called `expectedLiteral` that contains the literal value as we expect it to be (that is, in case of nested tags with one added xmlns on the outer tag, and in case of two tags that are not nested, both having the xmlns set). 

Does that make it clearer?

**Comment by 29kritika** [comment]
> Are you sure this includes all the changes you need for [GH-543](https://github.com/eclipse/rdf4j/issues/543)? On your previous PR, there were some changes to the SyntaxTreeBuildertokenManager (see these two commits: [8143255](https://github.com/eclipse/rdf4j/commit/81432558ea2634eafae646f5e38846c35702338f) and [fcbdc71](https://github.com/eclipse/rdf4j/commit/fcbdc718c1af790f690bfb7fbf847e1a6c307ab4)) which don't appear here. Were those changes unnecessary, or did you overlook them?

[8143255](https://github.com/eclipse/rdf4j/commit/81432558ea2634eafae646f5e38846c35702338f) this commit was giving a merge conflict while cherry-picking, so instead of cherry picking we added the changes to a new commit.
The changes in `SyntaxTreeBuildertokenManager` were related to the other issue (#1105 ) that we were working on, so we removed those changes from this branch.

File: core/rio/rdfxml/src/main/java/org/eclipse/rdf4j/rio/rdfxml/SAXFilter.java [hunk 1]
```diff
@@ -558,13 +558,35 @@ private void insertUsedContextPrefixes() {
...
+						} else {
+							opentag--;
+						}
+					}
+				}
+				i += 1;
+			}
+
+			System.out.println("Charbuf after insert: ");
+			System.out.println(charBuf);
```
[the remark this answers is not in the document]
Comment (deepali17043) [reply core/rio/rdfxml/src/main/java/org/eclipse/rdf4j/rio/rdfxml/SAXFilter.java]: I've added a commit `3a2b84f0f16b2a20f77a770446a74a56a9f7a578 [https://github.com/eclipse/rdf4j/pull/2290/commits/3a2b84f0f16b2a20f77a770446a74a56a9f7a578]` removing these statements.

[the remark this answers is not in the document]
Comment (deepali17043) [reply core/rio/rdfxml/src/main/java/org/eclipse/rdf4j/rio/rdfxml/SAXFilter.java]: I've added a commit that removes these statements [https://github.com/eclipse/rdf4j/pull/2290/commits/3a2b84f0f16b2a20f77a770446a74a56a9f7a578](url)

File: core/rio/rdfxml/src/main/java/org/eclipse/rdf4j/rio/rdfxml/SAXFilter.java [hunk 2]
```diff
@@ -558,13 +558,35 @@ private void insertUsedContextPrefixes() {
 				String prefix = unknownPrefixesInXMLLiteral.get(i);
 				String namespace = topElement.getNamespace(prefix);
 				if (namespace != null) {
+					System.out.println(prefix);
+					System.out.println(namespace);
```
[the remark this answers is not in the document]
Comment (deepali17043) [reply core/rio/rdfxml/src/main/java/org/eclipse/rdf4j/rio/rdfxml/SAXFilter.java]: I've added a commit that removes these statements [3a2b84f](https://github.com/eclipse/rdf4j/pull/2290/commits/3a2b84f0f16b2a20f77a770446a74a56a9f7a578)

**Comment by 29kritika** [comment]
> > Including the RDFXML Parser Test requires the private class `CanonXMLValueFactory` and we are unable to access it in `RDFXMLWriterTest`
> > Please suggest what can be done.
> 
> Apologies, I may have confused things by suggesting you should add a test to RDFXMLWriterTest - that should of course be RDFXMLParserTest! After all, the fix you did is a fix in the parser, not in the writer.
> 
> So, to make it clear:
> 
> 1. add one or two new test methods to `org.eclipse.rdf4j.rio.rdfxml.RDFXMLParserTest`
> 2. in these methods, read a file that contains a value with parse type literal - there are some existing tests there that read a file, the actual files can be found in `src/test/resources`, so just add one or two new files there for your test cases.
> 3. parse this file with the RDFXMLParser, and then check  (using `assertTrue` or `assertEquals`) that the actual literal value that the parser produces is as expected. You can easily do this by just having a separate String called `expectedLiteral` that contains the literal value as we expect it to be (that is, in case of nested tags with one added xmlns on the outer tag, and in case of two tags that are not nested, both having the xmlns set).
> 
> Does that make it clearer?

Thanks. We have added a test to `org.eclipse.rdf4j.rio.rdfxml.RDFXMLParserTest.java`  and checked  using `assertEquals` that the actual literal value that the parser produces is as expected.

**Pull request merged**